[Improvement]: Display the title as CSV/XLSX header

SamyMP opened the pull request:
<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `10.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `10.5` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Resolves #14163 

## Additional info

SamyMP committed:
Apply php-cs-fixer changes

SamyMP committed:
Merge branch 'pimcore:11.x' into 11.x

SamyMP committed:
Display label as CSV/XLSX header export

SamyMP commented:
This PR is still WIP, I wanted to open a draft so it's not ready yet

blankse commented:
@SamyMP You can click to the "Convert to draft" link in the reviewers section in the sidebar.

SamyMP committed:
Display label as CSV/XLSX header export

SamyMP commented:
The PR should be ready for review now.

By default, the export will now use the label of the fields instead of its name (which also is its key) configured in the dataobject class. If there is no label (label is not mandatory), it falls back to its name (key)

fashxp commented:
@SamyMP thank you very much for you contribution. 
Could be make that configurable in the export dialog? Maybe even via a select box with the options `header name`, `header title`, `no header` ... default value to `header title`.

SamyMP committed:
Add option in dialog box to choose heade type

SamyMP commented:
@fashxp I made the modification you asked in your comment, it is now configurable. The last thing to do is to potentially remove the first line in case you choose "No Header". Currently, the first line is empty.
I'll try to solve this problem on Monday if no one has before

fashxp commented:
Nice, thx very much!

SamyMP committed:
Remove empty first line when no header selected

SamyMP committed:
Merge branch '11.x' into 14163-display-title-csv-xlsx-header

SamyMP commented:
Is there something I have to do about the tests failing ? Is it something I have the ability to do ?

NiklasBr commented on models/DataObject/Service.php:
```diff
@@ -1815,11 +1818,17 @@ protected static function mapFieldname(string $field, array $helperDefinitions):
+            return $key;
+        } elseif ($header === 'no_header') {
+            return '';
+        }
```
Maybe this part can be moved to the top of the method? Because it seems that it could be an early return with no need for further processing.

```php
if ($header === 'no_header') {
    return '';
}
```

SamyMP replied:
Indeed, thank you for pointing that out

SamyMP committed:
Move no_header condition to top of method

SamyMP committed:
Merge branch '14163-display-title-csv-xlsx-header' of github.com:SamyMP/pimcore into 14163-display-title-csv-xlsx-header

SamyMP committed:
Merge branch '11.x' into 14163-display-title-csv-xlsx-header

brusch commented:
@SamyMP see errors reported here: https://github.com/pimcore/pimcore/actions/runs/4133230525/jobs/7142884005#step:8:25 😊

SamyMP committed:
Merge branch 'pimcore:11.x' into 14163-display-title-csv-xlsx-header

SamyMP committed:
fix tests

SamyMP commented:
Thank you @brusch, tests are ok now 🙂

Corepex commented on bundles/AdminBundle/public/js/pimcore/object/gridexport/xlsx.js:
```diff
@@ -38,6 +38,26 @@ pimcore.object.gridexport.xlsx = Class.create(pimcore.element.gridexport.abstrac
         });
+    },
+    getExportSettingsContainer: function () {
+        console.log('test')
```
```suggestion
```

we should get rid of that ...

Corepex commented on bundles/CoreBundle/translations/admin.en.yaml:
```diff
@@ -1,3 +1,247 @@
+gdpr_dataSource_sentMail: Sent Emails
+email_log_subject: Subject
+gdpr_data_extractor_label_email: Email Address
+workflow_notes: Notes
+2fa_code: Code
+password: Password
+show_metainfo: Info
+email_log_from: From
+email_from: From
+log_search_from: From
+email_log_to: To
+email_to: To
+log_search_to: To
+structuredtable_type_text: Text
+objectsMetadata_type_text: Text
+gdpr_data_extractor_label_id: ID
+navigation_target: Target
+log_type: Type
+gdpr_dataSource_export: Export
+classificationstore_configuration: Configuration
+gdpr_dataSource_assets: Assets
+structuredtable_type_bool: Checkbox
+objectsMetadata_type_select: Select
+objectsMetadata_type_bool: Bool
+gdpr_data_extractor_label_firstname: Firstname
+gdpr_data_extractor_label_lastname: Lastname
+log_source: Source
+classificationstore_error_addgroup_title: Error
+classificationstore_error_addkey_title: Error
+element_lock_message: 'The desired element is currently opened by another person:'
+log_message: Message
+operator_substring_settings: Settings
+delete_gridconfig_dblcheck: Do you really want to delete this item?
+email_log_cc: Cc
+email_log_bcc: Bcc
+cols: Columns
+log_search_message: Message
+header: Header
+no_header: No Header
+batch_export_confirmation: You are about to export %s dataObjects. Do you want to continue ?
```
These translations have been moved to admin bundle (https://github.com/pimcore/pimcore/commit/736d45a95ac3a1ab2b067950a46d71b8f6bf5d35)

SamyMP committed:
Remove console.log in bundles/AdminBundle/public/js/pimcore/object/gridexport/xlsx.js

Co-authored-by: Christian F. <kontakt+github@christian-feldkirchner.at>

Corepex commented:
@SamyMP, thanks for your PR 🎉 export works as expected - just two small things need to be changed.
I'm, really looking forward to check this PR again 👍

SamyMP committed:
fix translations

SamyMP committed:
Merge branch '14163-display-title-csv-xlsx-header' of github.com:SamyMP/pimcore into 14163-display-title-csv-xlsx-header

SamyMP committed:
fix translations

SamyMP commented:
Hi @Corepex, thank you for your review, I moved the translation and remove the `console.log` (sorry about that!),
I moved the translations to the admin bundle as well 😊

Corepex commented:
@SamyMP, thanks for the quick patches 👍 Now everything looks good to me - thanks for contributing 🥇 🍭
